src/upload.ts

```diff
@@ -14,8 +14,26 @@ export async function upload (args: string[], internal = false) {
     : urlOrDirOrSqliteFile.endsWith('.db')
       ? uploadEntryToSQLite
       : uploadEntryToURL
-  for (const filepath of files) {
-    const entry = await createEntryFromFile(filepath)
+  for (const filepath_ of files) {
+    let type = multicodes.RAW
+    let filepath = filepath_
+    if (filepath_[1] === '|') {
+      switch (filepath_[0]) {
+        case 'r':
+          // raw file type
+          break
+        case 'm':
+          type = multicodes.SHELTER_CONTRACT_MANIFEST
+          break
+        case 't':
+          type = multicodes.SHELTER_CONTRACT_TEXT
+          break
+        default:
+          throw new Error('Unknown file type: ' + filepath_[0])
+      }
+      filepath = filepath_.slice(2)
+    }
+    const entry = await createEntryFromFile(filepath, type)
```

Comment:
Is it not a concern that a user might use `chel upload` on a file that has `|` as the 2nd character?

Reply:
Files can't have `|` on Windows (one of the reasons I chose it. However, for such files, they could be prefixed as needed. E.g., if the filename is `||`, one could use `r|||`

Reply:
On POSIX, only `0x00` (and maybe `/`) is an invalid character in file name, but using `0x00` for this purpose looked wrong, as it's difficult to type in a terminal.

Reply:
OK, then do you make sure that everywhere `upload` is called the passed in files are prefixed?

And if they're all prefixed, what's the point of check on line 20? Instead, we should probably throw an exception if `|` isn't found.

Reply:
The intent was to allow the command to be used from the command line without a prefix (in which case a raw id would be used)

Reply:
I think if it's used from the command line we can detect that by checking the `internal` variable, right?

If so, then you should be able to throw an exception when `internal = true` and we don't have `|`, and when `internal = false` we can add the prefix manually inside of the `upload` function before the `switch` statement hits.

Reply:
But what if the user wishes to set the prefix themselves?

Reply:
Then we can invent some other flag or function for that.

Reply:
But a flag wouldn't work very well because the `upload` command works with multiple files.

Reply:
Sure, but I don't really see the ambiguity here as `|` is pretty rare as a file name, and I expect that having a file called `r|XX` and another one called `XX` is even rarer.

Reply:
I agree, it's rare, but not so rare that it never happens. On macOS `|` is a valid filename character. If we can fix this now and forget about it, we don't need to ever worry about it in the future.